### PR TITLE
Unit List View

### LINKED ISSUE #93: Advanced Unit Browsing - Units List View
# Overview

Implement the list view for the home page.

[Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=2779%3A8766&mode=dev)

## Frontend

First, implement the card/list view toggle button at the top of the page:

image

Next, implement the list view for units:

image

The data for the units lives in the `units` `useState` so you just need implement the different view for displaying the unit data.

## Backend

n/a

## Acceptance Criteria
- [x] List/Card view can be toggled via button at top of page
- [x] List view implemented

### PR DESCRIPTION
**Pull request by PoliteUnicorn**
## Tracking Info

Resolves #93

## Changes
Implemented the card/list view toggle button at the top of the page and the list view for units.
Also replaced paginator with V2 button version. 

## Testing
Tested by switching between unit and list views and trying the paginator on varying numbers of pages. 

## Confirmation of Change
The implemented list view: 
Screenshot 2024-05-22 at 1 03 12 PM
Screenshot 2024-05-22 at 1 03 19 PM

**The paginator:** 
Screenshot 2024-05-22 at 1 03 42 PM
Screenshot 2024-05-22 at 1 03 54 PM
Screenshot 2024-05-22 at 1 04 00 PM
Screenshot 2024-05-22 at 1 04 06 PM

### CODE REVIEW

File: frontend/public/Group.svg
Comment (petabite) [inline frontend/public/Group.svg:1]: Looks like this file isn't used anywhere

File: frontend/src/components/ReferralTable.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { useContext, useState } from "react";
 import { Link } from "react-router-dom";
 import styled from "styled-components";
+// import { Pagination } from "./Pagination";
```
Comment (petabite) [inline frontend/src/components/ReferralTable.tsx]:
```suggestion
```
Can we remove this?

File: frontend/src/components/Table.tsx [hunk 2]
```diff
@@ -82,7 +85,7 @@ export const Table = (props: TableProps) => {
             ))}
           <TableFooter>
             {/* TODO: Replace this with new pagination when it is implemented */}
-            <TablePagination
+            <Pagination
```
Comment (petabite) [inline frontend/src/components/Table.tsx:88]: Thanks for porting over the new pagination here!

**Review by petabite — CHANGES_REQUESTED** [review]
This looks awesome! Left some small comments below. Also, do you think it is possible to vertically center the content in each list item?

image

File: frontend/public/Group.svg
Comment (PoliteUnicorn) [reply frontend/public/Group.svg:1]: Deleted!

**Review by PoliteUnicorn — COMMENTED** [review]
Fixed changes!

**Review by petabite — CHANGES_REQUESTED** [review]
It looks like the row content isn't vertically centered yet.

Also, do you think you can fix the spacing of the "No matching units found" text here?

image

**Comment by PoliteUnicorn** [comment]
Fixed alignment issues and text.

**Review by petabite — APPROVED** [review]
Nice work!! 🚢

**Review by RamtinTJB — APPROVED** [review]
Great work! 💯 🗿

**Pull request merged**